standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -204,7 +218,8 @@ public CompactionInfo findNextToCompact(String workerId, String workerVersion) t
           info.properties = rs.getString(6);
           // Now, update this record as being worked on by this worker.
           long now = getDbTime(dbConn);
-          s = "UPDATE \"COMPACTION_QUEUE\" SET \"CQ_WORKER_ID\" = '" + workerId + "', \"CQ_WORKER_VERSION\" = '" + workerVersion + "', " +
+          s = "UPDATE \"COMPACTION_QUEUE\" SET \"CQ_WORKER_ID\" = '" + rqst.getWorkerId() + "', " +
+            "\"CQ_WORKER_VERSION\" = '" + rqst.getWorkerVersion() + "', " +
```

Comment:
I believe the `getWorkerVersion()` can be null, if this method is called from `findNextToCompact(String workerId)`. Is `CQ_WORKER_VERSION='null'` is a valid scenario? Are we prepared to handle null values when processing the `CQ_WORKER_VERSION` column?

Reply:
Great question. It turns out we do handle this case, we filter for: Objects::nonNull when counting the number of versions. However this case wasn't covered in tests so I'll update the PR with coverage.